app/multitenant/dynamo_collector.go

```diff
@@ -330,9 +375,27 @@ func (c *dynamoDBCollector) getReports(userid string, row int64, start, end time
 		return nil, err
 	}
 
+	// TODO(jml): Refactor this so that we have a standard interface
+	// IReportStore for "fetch :: [ReportKeys] -> ([Report], missing, error)"
+	// and a list of providers: in-memory cache, memcache, dynamo. Possibly
+	// also an "always everything missing" implementation to simplify logic.
```

Comment:
@tomwilkie By "break out the type", do you mean define a named struct type & use that rather than a tuple? (I think that's a good idea)